src/main/java/gary/Gary.java

```diff
@@ -0,0 +1,183 @@
+package gary;
+
+import gary.exception.*;
+import gary.task.Deadline;
+import gary.task.Event;
+import gary.task.Task;
+import gary.task.Todo;
+
+import java.util.Scanner;
+
+public class Gary {
+    public static final int MAX_TASK = 100;
+
+    public static final int TODO_DESCRIPTION_START_INDEX = 5;
+    public static final int DEADLINE_DESCRIPTION_START_INDEX = 9;
+    public static final int DEADLINE_BY_SPACE_LENGTH = 4;
+    public static final int EVENT_DESCRIPTION_START_INDEX = 6;
+    public static final int EVENT_FROM_SPACE_LENGTH = 6;
+    public static final int EVENT_TO_SPACE_LENGTH = 4;
+    public static void main(String[] args) {
+        Scanner in = new Scanner(System.in);
+        greetings();
+
+        Task[] todos = new Task[MAX_TASK];
+        int todosCount = 0;
+
+        String line;
+        line = in.nextLine();
+        String[] lineWords;
+        String command;
+
+        while (!(line.toUpperCase().contains("BYE"))) {
+            lineWords = line.split(" ");
+            command = lineWords[0];
+            if (line.equalsIgnoreCase("LIST")) {
+                processList(todosCount, todos);
+            } else if (command.equalsIgnoreCase("MARK")) {
+                processMark(todos, lineWords);
+            } else if (command.equalsIgnoreCase("UNMARK")) {
+                processUnmark(todos, lineWords);
+            } else {
+                try {
```

Comment:
You could consider refactoring this method to avoid deep nesting to adhere to the code quality [guideline](https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-deep-nesting) and to avoid long methods ([guideline](https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods))